docs/standard/microservices-architecture/microservice-ddd-cqrs-patterns/implement-value-objects.md

```diff
@@ -120,25 +122,170 @@ public class Address : ValueObject
+
+In the initial version of eShopOnContainers (.NET Core 1.1), the hidden ID needed by EF Core infrastructure was implemented in the following way in the DbContext level, using Fluent API at the infrastructure project. Therefore, the ID was hidden from the domain model point of view, but still present in the infrastructure.
+
+```csharp
+// Old approach with EF Core 1.1
+// Fluent API within the OrderingContext:DbContext in the Infrastructure project
+void ConfigureAddress(EntityTypeBuilder<Address> addressConfiguration) 
+{
+    addressConfiguration.ToTable("address", DEFAULT_SCHEMA); 
+
+    addressConfiguration.Property<int>("Id")  // Id is a shadow property
+        .IsRequired();
+    addressConfiguration.HasKey("Id");   // Id is a shadow property
+}
+```
+
+However, the persistence of that value object into the database was performed like a regular entity in a different table.
+
+With EF Core 2.0 there are new and better ways to persist value objects.
+
+## Persist value objects as owned entity types in EF Core 2.0
+
+Even when there are still some gaps between the canonical value object pattern in DDD and the owned entity type in EF Core (see limitations at the end of this section), it is currently, with EF Core 2.0, the best way to persist value objects.
+
+The onwed entity type feature was added to EF Core since version 2.0.
+
+An owned entity type allows you to map types that do not have their own identity explicitely defined in the domain model and are used as properties (such as a value object) within any of your entities. An owned entity type shares the same CLR type with another entity type. The entity containing the defining navigation is the owner entity. When querying the owner, the owned types will be included by default.
+
+Just by looking at the domain model, an owned type looks like it doesn’t have any identity but, under the covers, owned types do have identity, but the owner navigation property is part of this identity.
+
+The identity of instances of own types is not completely their own. It consists of three components: they identity of the owner, the navigation property pointing to them, and in the case of collections of owned types (not supported yet in EF Core 2.0), an independent component.
```

Comment:
Typo: they identity

I recommend breaking the last sentence into bullet items and the last item can have the flow improved by moving the not supported info to the end 

not supported yet -> not yet supported

Reply:
👍